### ISSUE TITLE
Secrets don't compare domain only?

### ISSUE THREAD

**Issue by mikeyyuen** [keel-hq/keel#301]
Should the `domainOnly(...)` check used in the function below apply to both the image url and to the url stored in the secret for the registry?

https://github.com/keel-hq/keel/blob/2175a981a38ceaa0e3dbcd667a31ff851b103788/secrets/secrets.go#L250

I've noticed if the registry address has a port in it then keel will not pick up the secret correctly.

**Comment by rusenask** [comment]
Hi, I think the bug that made it not recognize registries with ports was fixed some time ago. Did you encountered a problem with a registry that has a port?

**Comment by mikeyyuen** [comment]
I think that PR fixes some issues but creates a new issue. For example, if the registry is stored in the secret with the following pattern

```
{"auths:{"foo:5000":{....}}}
```

And a deployment controller specifies an image say `foo:5000/myimage` then the code above will extract the image domain as `foo` from the image and then look in the secret for a matching registry named `foo` which it won't find as the registry has the port attached in the secret.

I did a quick check, if you `docker login foo:5000` the `.docker/config.json` will have an auths section that follows the above format (i.e. port included).

I was able to fix this my end by adding a two entries to the secret (one with and one without the port), but I think this may be an un-intentional bug / regression?
